cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -483,13 +483,17 @@ func (m *recoverModeStateMachine) verifyMountPoint(dir, name string) error {
 	return nil
 }
 
-func (m *recoverModeStateMachine) setFindState(partName, partUUID string, err error) error {
+func (m *recoverModeStateMachine) setFindState(partName, partUUID string, err error, optionalPartition bool) error {
 	part := m.degradedState.partition(partName)
 	if err != nil {
 		if _, ok := err.(disks.PartitionNotFoundError); ok {
-			// explicit error that the device was not found
-			part.FindState = partitionNotFound
-			m.degradedState.LogErrorf("cannot find %v partition on disk %s", partName, m.disk.Dev())
+			if optionalPartition {
+				part.FindState = partitionAbsentOptional
```

Comment:
below MountState was set but here it's FindState, is that a problem? I see later that we set MountState too, but still this feels like a change to the content of degraded?

Reply:
Ah right, forgot to clean this one up before committing. Let me push a quick patch